lib/Files/Util.php

```diff
@@ -110,6 +144,31 @@ public function getAnyDirectShareOfFileAccessibleByUser(string $fileId, string $
 		return null;
 	}
 
+	/**
+	 * Returns any public share of the node (like a link share) created by the
+	 * user.
+	 *
+	 * @param Node $node
+	 * @param string $userId
+	 * @return IShare|null
+	 */
+	private function getAnyPublicShareOfNodeOwnedByUser(Node $node, string $userId): ?IShare {
+		$reshares = false;
+		$limit = 1;
+
+		$shares = $this->shareManager->getSharesBy($userId, \OCP\Share::SHARE_TYPE_LINK, $node, $reshares, $limit);
+		if (\count($shares) > 0) {
```

Comment:
Dont count when you dont need the number:
```suggestion
		if (!empty($shares)) {
```